ompi/mca/osc/monitoring/configure.m4

```diff
@@ -0,0 +1,89 @@
+# -*- shell-script -*-
+#
+# Copyright (c) 2016-2017 Inria.  All rights reserved.
+# $COPYRIGHT$
+#
+# Additional copyrights may follow
+#
+# $HEADER$
+#
+
+# mca_ompi_osc_monitoring_generate_templates
+#
+# Append to file $1 an array of components by listing component names in $2.
+#
+# $1 = filename
+# $2 = osc component names
+#
+AC_DEFUN([mca_ompi_osc_monitoring_generate_templates],[
+    m4_ifval([$1], [
+        cat <<EOF >[$1]
+/*
+ * DO NOT EDIT - This file generated from
+ * ompi/mca/osc/monitoring/configure.m4
+ */
+/*
+ * Copyright (c) 2017      Inria.  All rights reserved.
+ * \$COPYRIGHT$
+ *
+ * Additional copyrights may follow
+ *
+ * \$HEADER$
+ */
+
+#ifndef MCA_OSC_MONITORING_GEN_TEMPLATE_H
+#define MCA_OSC_MONITORING_GEN_TEMPLATE_H
+
+#include <string.h>
+#include <ompi_config.h>
+#include <ompi/mca/osc/osc.h>
+#include "osc_monitoring_template.h"
+
+/***************************************/
+/* Include template generating macros  */
+
+EOF
+        # Generate each case in order to register the proper template functions
+        for comp in `echo $2 | sed -e 's/,/ /g' -e 's/monitoring//'`
+        do
+            echo "OSC_MONITORING_MODULE_TEMPLATE_GENERATE(${comp})" >>[$1]
+        done
+        cat <<EOF >>[$1]
+
+/***************************************/
+
+static inline int
+ompi_mca_osc_monitoring_set_template(ompi_osc_base_component_t *best_component,
+                                     ompi_osc_base_module_t *module)
+{
+EOF
+        echo -n '    ' >>[$1]
+        for comp in `echo $2 | sed -e 's/,/ /g' -e 's/monitoring//'`
+        do
+            echo "if ( 0 == strcmp(\"${comp}\", best_component->osc_version.mca_component_name) ) {" >>[$1]
+            echo "        OSC_MONITORING_SET_TEMPLATE(${comp}, module);" >>[$1]
+            echo -n '    } else ' >>[$1]
+        done
+        echo '' >>[$1]
+        cat <<EOF >>[$1]
+        return OMPI_ERR_NOT_SUPPORTED;
+    return OMPI_SUCCESS;
+}
+
+#endif /* MCA_OSC_MONITORING_GEN_TEMPLATE_H */
+EOF
+    ])dnl
+])dnl
+
+# MCA_ompi_osc_monitoring_CONFIG()
+# ------------------------------------------------
+AC_DEFUN([MCA_ompi_osc_monitoring_CONFIG],[
+    AC_CONFIG_FILES([ompi/mca/osc/monitoring/Makefile])
+
+    AS_IF([test "$MCA_BUILD_ompi_common_monitoring_DSO_TRUE" = ''],
+          [$1],
+          [$2])
+
+   mca_ompi_osc_monitoring_generate_templates([$OMPI_TOP_SRCDIR/ompi/mca/osc/monitoring/osc_monitoring_template_gen.h],
```

Comment:
I'm surprised this worked with "make distcheck"; you should be writing into the build directory, not the source directory.  You actually shouldn't need the prefix (`$OMPI_TOP_SRCDIR/`) at all; configure is by definition run at the top of the list.

Reply:
It did not. I think the cleancheck failed (but I did not know that at that point. I fought latter with targets to make it succeed. And I reached beyond despair at some point... But now it seems fine)